tests/test_utils.py

```diff
@@ -41,11 +41,11 @@ class TestUtils:
     @pytest.mark.parametrize(
         "input_bytes,byte_pattern,pattern_mask,lstrip,expected",
         [
-            (input_bytes, b"GIF87a", b"\xff\xff\xff\xff\xff\xff", None, True),
-            (input_bytes, b"GIF87a", b"\xff\xff\xff\xff\xff", None, ValueError),
-            (b" \t\n\rGIF87a", b"GIF87a", b"\xff\xff\xff\xff\xff\xff", WHITESPACE_BYTES, True),
-            (b"GIF", b"GIF87a", b"\xff\xff\xff\xff\xff\xff", None, False),
-            (b"\xff\xff\xff\xff\xff\xff", b"GIF87a", b"\xff\xff\xff\xff\xff\xff", None, False),
+            (input_bytes, b"GIF87a", bytes.fromhex("ffffffffffff"), None, True),
+            (input_bytes, b"GIF87a", bytes.fromhex("ffffffffff"), None, ValueError),
+            (b" \t\n\rGIF87a", b"GIF87a", bytes.fromhex("ffffffffffff"), WHITESPACE_BYTES, True),
+            (b"GIF", b"GIF87a", bytes.fromhex("ffffffffffff"), None, False),
+            (bytes.fromhex("ffffffffffff"), b"GIF87a", bytes.fromhex("ffffffffffff"), None, False),
```

Comment:
You could just leave the strings here and do `value if isinstance(value, bytes) else bytes.fromhex(value)` in the text code.

Reply:
In that case, how should I handle concatenation of strings like this
 `bytes.fromhex("000000") + b" ftyp2222" + bytes.fromhex("00000000") + b"2222mp41isomavc1"`?

Reply:
> You could just leave the strings here and do `value if isinstance(value, bytes) else bytes.fromhex(value)` in the text code.

Oh, you mean particularly for this test, right?

Reply:
For concatenation, you could make the source a tuple of bytes and strings, and concatenate conversion included in the test, e.g.

```
("000000", b" ftyp2222", "00000000", b"2222mp41isomavc1")
```

then

```
b''.join(value if isinstance(value, bytes) else bytes.fromhex(value) for value in values)
```

Also, in cases where there is only 1 readable character, it may be better to still use hexadecimal for the whole thing. For example:

```
bytes.fromhex("1a") + b"E" + bytes.fromhex("dfa3") + b"B" + bytes.fromhex("82")
```

would probably be more readable as

```
"1a 45 df a3 42 82"
```

(note the spaces as well, as @elacuesta suggested earlier)